solutions/src/main/java/ru/sberbank/school/task02/ReversedCurrencyCalc.java

```diff
@@ -0,0 +1,323 @@
+    }
+
+    public Optional<BigDecimal> convertReversed(@NonNull ClientOperation operation,
+                                                @NonNull Symbol symbol,
+                                                @NonNull BigDecimal amount,
+                                                double delta,
+                                                @NonNull Beneficiary beneficiary) {
+
+        if (amount.compareTo(BigDecimal.ZERO) <= 0) {
+            throw new IllegalArgumentException("amount should be positive");
+        }
+
+        List<Quote> quotes = externalQuotesService.getQuotes(symbol);
+        if (quotes == null || quotes.isEmpty()) {
+            throw new FxConversionException("no available quotes");
+        }
+
+        SortedSet<ReversedQuote> prices = findPrices(quotes, operation, amount);
+
+        return getPrice(prices, beneficiary, operation);
+    }
+
+    /**
+     * This function picks two quotes with volumes
+     * that have bounds closest to amount.
+     * @param quotes List of available quotes.
+     * @param direction Operation type.
+     * @param amount Amount in target currency
+     * @return Pair of reversed quotes wich are made
+     *          from two suitable quotes.
+     */
+    private SortedSet<ReversedQuote> findPrices(@NonNull List<Quote> quotes,
+                                                 @NonNull ClientOperation direction,
+                                                 @NonNull BigDecimal amount) {
+
+        SortedSet<ReversedQuote> prices = new TreeSet<>(Comparator.comparing(ReversedQuote::getPrice));
+        Iterator<Quote> iter = quotes.iterator();
+
+        ReversedQuote left = new ReversedQuote(iter.next(), direction);
+        ReversedQuote right = new ReversedQuote(iter.next(), direction);
+        ReversedQuote current;
+
+        Volume leftVolume = left.getQ().getVolume();
+        Volume rightVolume = right.getQ().getVolume();
+        Volume currentVolume;
+
+        if (compare(rightVolume, leftVolume) < 0) {
+
+            current = right;
+            right = left;
+
+            currentVolume = rightVolume;
+            rightVolume = leftVolume;
+            leftVolume = currentVolume;
+            left = current;
+        }
+
+        right.setLowerBound(leftVolume);
+        Position currentPos = checkBounds(left.getLowerBound(), left.getUpperBound(),
+                                          right.getLowerBound(), right.getUpperBound(), amount);
+
+        if (currentPos == Position.INTERSECTION) {
+
+            prices.add(left);
+            prices.add(right);
+        } else if (currentPos == Position.LEFT) {
+            prices.add(left);
+        } else if (currentPos == Position.RIGHT) {
+            prices.add(right);
+        }
+
+        while (iter.hasNext()) {
+
+            current = new ReversedQuote(iter.next(), direction);
+            currentVolume = current.getQ().getVolume();
+
+            if (compare(currentVolume, leftVolume) >= 0 &&
```

Comment:
&#39;&amp;&amp;&#39; should be on a new line.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-109448826>*